x-pack/plugins/ml/server/lib/register_settings.ts

```diff
@@ -30,5 +34,43 @@ export function registerKibanaSettings(coreSetup: CoreSetup) {
         }),
       },
     },
+    [ANOMALY_DETECTION_ENABLE_TIME_RANGE]: {
+      name: i18n.translate('xpack.ml.advancedSettings.enableAnomalyDetectionDefaultTimeRangeName', {
+        defaultMessage: 'Enable time filter defaults for anomaly detection jobs',
+      }),
+      value: false,
+      schema: schema.boolean(),
+      description: i18n.translate(
+        'xpack.ml.advancedSettings.enableAnomalyDetectionDefaultTimeRangeDesc',
+        {
+          defaultMessage:
+            'Use the default time filter in the Single Metric Viewer and Anomaly Explorer.',
+        }
+      ),
+      category: ['Machine Learning'],
+    },
+    [ANOMALY_DETECTION_DEFAULT_TIME_RANGE]: {
+      name: i18n.translate('xpack.ml.advancedSettings.anomalyDetectionDefaultTimeRangeName', {
+        defaultMessage: 'Time filter defaults',
+      }),
+      type: 'json',
+      value: `{
```

Comment:
is it worth pulling these defaults out into constants like our existing file data viz setting.
this could be stored as a real json object constant and stringified when adding it here.
`JSON.stringify(FOO, null, 2)` for the correct formatting.

Reply:
This is super neat! Updated here https://github.com/elastic/kibana/pull/76347/commits/1118b3e08345110bec102f86bfbd3bf46cce969c

Although I put the new settings in the `settings` constants instead of creating a new file because I didn't think it was big enough. Although I'm happy to create a new file for them!